### ISSUE TITLE
may i combine the classification with NMT

### ISSUE THREAD

**Issue by 520jefferson** [juliakreutzer/quetch#1]
@juliakreutzer 
nice to meet you julia , i search with google and  see you page: https://statnlp.github.io/rl4nmt/#
After classification i can get a signal to tell whether the translation result  is good or bad.

how to combine the classification model with NMT training? 
maybe we can add the classification result as a feature or modify the loss function, is there any other way?

**Comment by juliakreutzer** [comment]
Hi @520jefferson 
Yes, a good quality estimation (QE) model can serve as a reward model for reinforcing good translations, that means by weighting outputs of your current NMT model with e.g. policy gradient or minimum-risk training or adversarial training.
The alternative approach to incorporate it as features for a second-pass translation model is definitively interesting as well - I have seen it applied for post-editing.
However, it seems challenging to achieve the necessary quality of the QE model in order to not misguide the NMT system.
Other options are multi-task learning or transfer learning: jointly learn QE and NMT or use NMT representations for QE.

**Comment by 520jefferson** [comment]
Hi @juliakreutzer 
Thanks for your answer!
How about finetune the NMT model like this below.
Fristly, i train the NMT model, and get the BLEU best model, then i get the vocabulary and the word embedding from the best NMT model.
Secondly, i train the classification model with the pretrianed word embedding and vocabulary.
Thirdly, i will fix the word embedding and use the classification to adjust the NMT traing by  union the loss of both classification and  NMT training.  On the other way, the classfication can generate a reliable result to show whether the result conforms to the specification , maybe it can serve as a reward model for reinforcing good translations by adjust the pretrained NMT model.

**Comment by juliakreutzer** [comment]
Sounds interesting :smile:
